File: VDFEditor/src/main/java/de/treichels/hott/vdfeditor/ui/LoadVoiceFileTask.java
package de.treichels.hott.vdfeditor.ui;

import de.treichels.hott.HoTTSerialPort;
import gde.model.voice.VoiceFile;

final class LoadVoiceFileTask extends TransmitterTask {
    private final boolean user;

    public LoadVoiceFileTask(final String title, final boolean user) {
        super(title);
        this.user = user;
    }

    @Override
    protected VoiceFile call() throws Exception {
        try (HoTTSerialPort port = getPort()) {
            port.open();
            port.turnRfOutOff();
            port.delay();
            return port.loadVoiceFile(user, this);
        }
    }
}

File: VDFEditor/src/main/java/de/treichels/hott/vdfeditor/ui/transmitter/LoadVoiceFileTask.kt
package de.treichels.hott.vdfeditor.ui.transmitter

import gde.model.HoTTException
import gde.model.voice.VoiceFile

internal class LoadVoiceFileTask(title: String, private val user: Boolean) : TransmitterTask(title) {
    override fun call(): VoiceFile? = port?.use { port ->
        port.open()
        port.turnRfOutOff()
        port.delay()

        if (user && port.getVoiceInfo(false).voiceVersion == 2000) throw HoTTException("NoUerVoiceFiles")

        return port.loadVoiceFile(user, this)
    }
}
